week1-final(amin)/app/src/main/java/com/kodeco/android/coordplot/MainActivity.kt

```diff
@@ -0,0 +1,121 @@
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.Slider
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.kodeco.android.coordplot.ui.theme.MyApplicationTheme
+
+class MainActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            MyApplicationTheme {
+                PlotSurface()
+            }
+        }
+    }
+}
+
+@Composable
+fun PlotSurface() {
+    Surface(
+        modifier = Modifier.fillMaxSize(),
+        color = Color.White
+    ) {
+        var xValue by remember { mutableStateOf(0.5f) }
+        var yValue by remember { mutableStateOf(0.5f) }
+
+        Column (
+            verticalArrangement = Arrangement.SpaceEvenly,
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Text(" Coordinate Plotter ",
+                fontSize = (36.sp)
+
+            )
+
+            Map(
+                xPercent = xValue,
+                yPercent = yValue,
+            )
+            /*side to side slider */
+            Slider(
+                value = xValue,
+                valueRange = 0.01f..1f,
+                onValueChange = { value ->
+                    xValue = value
+                }
+            )
+            /* up and down slider */
+            Slider(
+                value = yValue,
+                valueRange = 0.01f..1f,
+                onValueChange = { value ->
+                    yValue = value
+                }
+            )
+        }
+    }
+}
+
+@Preview(showBackground = true)
+@Composable
+fun PlotSurfacePreview() {
+    MyApplicationTheme {
+        PlotSurface()
+    }
+}
+
+@Composable
+fun Map(xPercent: Float, yPercent: Float, modifier: Modifier = Modifier,
+        /*value: Float = 0.5f,
+        valueChanged: (Float) -> Unit */
+){
+    Box(
+        modifier = Modifier
+            .size(300.dp)
```

Comment:
For this value you could also go one step further and create `private const val`s outside of the function so that way these have names associated with the values:

```kotlin
private const val OUTSIDE_BOX_SIZE = 300
private const val INSIDE_BOX_SIZE = 36
private val INSIDE_BOX_RADIUS = INSIDE_BOX_SIZE / 2
```